src/preload/index.ts

```
@@ -65,6 +65,7 @@ import {
   CONFIG_UPDATE_TRIGGER,
 } from './constants/ipcChannels';
 
+import type { SessionAppendEvent, TerminalStateChangeEvent } from '@main/types';
```

ESLint

[error] No rule allowing this dependency was found. File is of type 'preload'. Dependency is of type 'main'

(boundaries/element-types)